Tests/Tests/CDTQIndexManagerEncryptionTests.m

```diff
@@ -0,0 +1,111 @@
+    // Put setup code here. This method is called before the invocation of each test method in the
+    // class.
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the
+    // class.
+    [super tearDown];
+}
+
+- (void)testCreateQueryIndexManagerWithEncryptionKeyNilProvider
+{
+    CDTEncryptionKeyNilProvider *provider = [CDTEncryptionKeyNilProvider provider];
+    CDTDatastore *datastore = [self.factory datastoreNamed:@"create_query_index_tests_nilprovider"
+                                 withEncryptionKeyProvider:provider
+                                                     error:nil];
+
+    NSError *err = nil;
+    CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:&err];
+
+    XCTAssertNotNil(im, @"indexManager is not nil");
+    XCTAssertNil(err, @"error has to be nil");
+}
+
+- (void)testCreateQueryIndexManagerWithEncryptionKeyNilProviderDoesNotCipherIndex
+{
+    // Create index
+    CDTEncryptionKeyNilProvider *provider = [CDTEncryptionKeyNilProvider provider];
+    CDTDatastore *datastore =
+        [self.factory datastoreNamed:@"create_query_index_tests_nilprovider_notcipher"
+            withEncryptionKeyProvider:provider
+                                error:nil];
+
+    __unused CDTQIndexManager *im =
+        [[CDTQIndexManager alloc] initUsingDatastore:datastore error:nil];
+
+    // Check
+    NSString *path = [CloudantSyncTests pathForQueryIndexInDatastore:datastore];
+
+    XCTAssertEqual([FMDatabase isDatabaseUnencryptedAtPath:path],
+                   kFMDatabaseUnencryptedIsUnencrypted,
+                   @"No encryption library available, database can not be encrypted");
+}
+
+- (void)testCreateQueryIndexManagerWithEncryptionKeyNilProviderFailsIfDBExistsAndItIsEncrypted
+{
+    // Create datastore
+    CDTEncryptionKeyNilProvider *provider = [CDTEncryptionKeyNilProvider provider];
+    CDTDatastore *datastore =
+        [self.factory datastoreNamed:@"create_query_index_tests_nilprovider_fails_with_cipher_db"
+            withEncryptionKeyProvider:provider
+                                error:nil];
+
+    // Create index
+    CDTQIndexManager *im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:nil];
+
+    // Replace index
+    NSFileManager *defaultManager = [NSFileManager defaultManager];
+    NSString *indexPath = [CloudantTests pathForQueryIndexInDatastore:datastore];
+
+    [defaultManager removeItemAtPath:indexPath error:nil];
+
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    NSString *assetPath = [bundle pathForResource:@"emptyencryptedindex" ofType:@"sqlite"];
+
+    [[NSFileManager defaultManager] copyItemAtPath:assetPath toPath:indexPath error:nil];
+
+    // Test
+    NSError *err = nil;
+    im = [[CDTQIndexManager alloc] initUsingDatastore:datastore error:&err];
+
+    XCTAssertNil(im, @"indexManager is nil");
+    XCTAssertNotNil(err, @"There is an error");
+}
```

Comment:
It is not possible to implement that test.

Check [this code](https://github.com/cloudant/CDTDatastore/commit/8ee69974ee935f7dbe3113b110de837916edf805#diff-6f72c4ee8250b96772fe8bc450cae5cbR72):
- A TD_Database can not be opened with a provider that returns a key
- A Datastore can not be created if a TD_Database can not be opened
- An index manager can not be created without a datastore.